### PR TITLE
Updated testing landing page, restructure via dirs

### PR DESCRIPTION
**Pull request by Nintendot**
## Overview

Merging testing wiki from g-testing to RA, update landing page, put docs in respective dirs
This is step 1 of merging to avoid massive code review at once.

### Features

- edited landing page

---

#### Meta

- [x] provide a descriptive topic
- [x] provide an overview of contribution
- [x] no sensitive content included, such as:
  - security & privacy policy violating content
  - content considered competitive intelligence
  - keys, tokens or credentials
- [x] documentation format follows [this template][template]
- [x] "work in progress" commits are squashed 
- [x] commits follow the [Karma][karma-format] or [Commitizen](https://www.npmjs.com/package/commitizen) format

[template]: ../.template.md
[karma-format]: https://karma-runner.github.io/1.0/dev/git-commit-msg.html

### CODE REVIEW

File: testing/README.md [hunk 1]
```diff
@@ -1,12 +1,67 @@
...
-- [Visual regression testing](visual-regression.md)
-- [Accessibility testing](accessibility.md)
-- [Load testing](load.md)
-- [SEO testing](seo.md)
-- [Analytics testing](analytics.md)
-- [Consumer driven contract testing](consumer_driven_contracts.md)
-- [Performance Testing](automated-testing.md)
+<img src="https://i.imgur.com/hpDNFJy.png" width=320 height=240>
+
+## Testing (in an RA world)
```
Comment (ahmadnassri) [inline testing/README.md]: this is implicit, don't need to mention "RA"

File: testing/README.md [hunk 2]
```diff
@@ -1,12 +1,67 @@
...
-- [End to End testing](e2e.md)
-- [Security testing](security.md)
-- [Visual regression testing](visual-regression.md)
-- [Accessibility testing](accessibility.md)
-- [Load testing](load.md)
-- [SEO testing](seo.md)
-- [Analytics testing](analytics.md)
-- [Consumer driven contract testing](consumer_driven_contracts.md)
-- [Performance Testing](automated-testing.md)
+<img src="https://i.imgur.com/hpDNFJy.png" width=320 height=240>
```
Comment (ahmadnassri) [inline testing/README.md]: lets avoid using graphics, since this will also be the source for the website.

Comment (Nintendot) [reply testing/README.md]: removed

File: testing/README.md (hunk 1 above)
Comment (Nintendot) [reply testing/README.md]: removed

File: testing/README.md [hunk 3]
```diff
@@ -1,12 +1,132 @@
...
+
+Here we lay out all the testing practices teams should be following in the RA world, whether they are automated, manually performed, [functional][Functional testing] or [non-functional][Non-functional testing]
+
+#### *Automated functional tests*
+These are tests that directly validates the functionality of your software / application.
+
+
+| What | Why | Who | When |
+| ---- | --- | --- | ---- |
+| [Unit tests][Unit tests] |[Why][Unit tests:why]| Developers | [When][Unit tests:when] |
```
Comment (ahmadnassri) [inline testing/README.md]: lets not use a table here, just use a list.

File: testing/README.md [hunk 4]
```diff
@@ -1,12 +1,132 @@
...
+---
+
+#### *Automated non-functional tests*
+These are tests that does not directly validate the functionality of your application. 
+
+However, you should invest in these tests, __because__ users’ affection and trust of a software or system is swayed and affected by these non-functional qualities. (Just imagine if Google takes 10 seconds to load your search results, or your Gmail password gets hacked every so often, would you trust their service still?)
+
+| What | Why | Who | Standards | When |
+| ---- | --- | --- | ---------------- | ------------- |
+| [SEO][SEO] | [Why][SEO:why] | Developers | [SEO standards][SEO:standards] | [When][SEO:when] |
```
Comment (ahmadnassri) [inline testing/README.md]: same as above

File: testing/README.md [hunk 5]
```diff
@@ -1,12 +1,132 @@
...
+[Unit tests]: functional/unit.md
+[Unit tests:why]: functional/unit.md#why
+[Unit tests:when]: functional/unit.md#when
+
+[Contract tests]: functional/consumer_driven_contracts.md
+[Contract tests:why]: functional/consumer_driven_contracts.md#illustrating-the-problem
+[Contract tests:when]: functional/consumer_driven_contracts.md#when
+
+[E2E functional]: functional/e2e.md
+[E2E functional:why]: functional/e2e.md#why
```
Comment (ahmadnassri) [inline testing/README.md]: too much detail, just link to the page and let the reader find their way through

File: testing/functional/consumer_driven_contracts.md [hunk 6]
```diff
@@ -22,3 +22,13 @@ You would then on a regular cadence (example: cron-job) make the actual request
 
 ## Testing from the provider side
 In an ideal scenario (when you have access to both the consumer and provider), the consumer would provide the contract to the provider. Then the provider would, in their pipeline, test against this contract. This makes sure that any code changes on the provider side does not break the contract they have with the consumer(s).
+
+## When
+
+Writing contract tests: During feature/story development
+Running contract tests: As part of the delivery pipeline
+
+## Standards
```
Comment (ahmadnassri) [inline testing/functional/consumer_driven_contracts.md]: empty?

Comment (Nintendot) [reply testing/functional/consumer_driven_contracts.md]: not changing every file, doing it incrementally so you have less to review at once

File: testing/README.md (hunk 3 above)
Comment (Nintendot) [reply testing/README.md]: done

File: testing/README.md (hunk 4 above)
Comment (Nintendot) [reply testing/README.md]: done

File: testing/README.md (hunk 5 above)
Comment (Nintendot) [reply testing/README.md]: done

**Pull request merged**